addon/v1alpha1/types_clustermanagementaddon.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Cluster"
+
+// ClusterManagementAddOn represents the registration of an add-on to the cluster management.
+// This resource allows the user to discover which add-on is available for the cluster management and
+// also provides metadata information about the add-on.
+// The resource also provides a linkage to ManagedClusterAddOn, the name of the ClusterManagementAddOn
+// resource will be used for the namespace-scoped ManagedClusterAddOn resource
+// ClusterManagementAddOn is a cluster-scoped resource.
+type ClusterManagementAddOn struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents a desired configuration for the agent on the cluster management add-on.
+	Spec ClusterManagementAddOnSpec `json:"spec"`
+
+	// Status represents the current status of cluster management add-on.
+	// +optional
+	Status ClusterManagementAddOnStatus `json:"status,omitempty"`
+}
+
+// ClusterManagementAddOnSpec provides the information of add-on CustomResourceDefinition.
+type ClusterManagementAddOnSpec struct {
+	// DisplayName represents the name that will be displayed.
+	// +required
+	DisplayName string `json:"displayName"`
+
+	// Description represents the detailed description of the add-on.
+	// +optional
+	Description string `json:"description"`
+
+	// AddOnConfigCRD is a reference to the name of the CRD that configures the add-on
+	// Note: There may be a case where a single add-on config CRD controls multiple related add-ons,
```

Comment:
do you meant there will be multiple clustermangeraddon resource that specify the same crd? Would you clarify the scenario?

Reply:
you are correct, multiple addon may share the same configuration resource and in this case the controller will create multiple clustermanageraddon

Reply:
updated the comment to further clarify the situation